src/compiler/xslt/invoke-compiled/invoke-compiled-current-scenario-or-expect.xsl

```diff
@@ -8,12 +8,10 @@
    <!--
       Generates an invocation of the template compiled from x:scenario or x:expect.
    -->
-   <xsl:template name="x:invoke-compiled-current-scenario-or-expect">
+   <xsl:template name="x:invoke-compiled-current-scenario-or-expect" as="element(xsl:call-template)">
       <!-- Context item is x:scenario or x:expect -->
       <xsl:context-item as="element()" use="required" />
 
-      <xsl:param name="last" as="xs:boolean" />
-
       <!-- URIQualifiedNames of the variables that will be passed as the parameters (of the same
          URIQualifiedName) to the compiled x:scenario or x:expect being invoked. -->
```

Comment:
This is tangential because these lines didn't change, but what does "(of the same URIQualifiedName)" mean?

Reply:
I think I wrote the phrase, but even I have difficulty understanding it. Probably it wanted to mean this:

"URIQualifiedNames of the variables" corresponds to Line 25.
"the parameters (of the same URIQualifiedName)" corresponds to Line 24:

https://github.com/xspec/xspec/blob/b977fea5a7d6e49048b841e96df3209705a3d26d/src/compiler/xslt/invoke-compiled/invoke-compiled-current-scenario-or-expect.xsl#L15-L28

"(of the same URIQualifiedName)" is trying to convey that `$with-param-uqnames` is used not only in Line 25 but also in Line 24. In particular "the same" is also trying to convey that `@select` in Line 24 is just `select="."` (not something like `select="'param-' || ."`).

On the other hand, in the XQuery version,

https://github.com/xspec/xspec/blob/b977fea5a7d6e49048b841e96df3209705a3d26d/src/compiler/xquery/invoke-compiled/invoke-compiled-current-scenario-or-expect.xsl#L16-L31

`$with-param-uqnames` is used only in Line 25.
There is no "(of the same URIQualifiedName)". Instead of it, the sequence order in `$with-param-uqnames` is vital important.

It would be appreciated if you could refine the source code comments based on those points :)

Reply:
Thanks for explaining. Now I get it. Here are some suggestions for the comment in the XSLT case:

`URIQualifiedNames of the variables that will be passed by name as the parameters to the compiled x:scenario or x:expect being invoked.` (shorter but doesn't convey that the names aren't being modified)

`URIQualifiedNames of the variables that will be passed, using their same URIQualifiedNames, as the parameters to the compiled x:scenario or x:expect being invoked.`

`URIQualifiedNames of the variables that will be passed as the parameters to the compiled x:scenario or x:expect being invoked. Names and contents of the variables are passed through unchanged.`

Reply:
Thanks @galtm. 
#1225 would be a better place because it handles `x:variable`. I took the third one of your ideas into 52faa90ec44bd893774782d53c9b4827f280f6a4 in #1225.